app/src/main/java/org/stacks/app/shared/ResultUtils.kt

```diff
@@ -1,12 +1,28 @@
 package org.stacks.app.shared
 
-import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.catch
-import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.*
 
 fun <T> Flow<T>.toResult(): Flow<Result<T>> =
     map { Result.success(it) }
         .catch { emit(Result.failure(it)) }
 
-fun <T, R> Flow<Result<T>>.mapIfSuccess(mapper: ((T) -> R)): Flow<Result<R>> =
-    map { it.map(mapper) }
+fun <T, R> Flow<Result<T>>.mapIfSuccess(mapper: suspend ((T) -> R)): Flow<Result<R>> =
+    map { result -> result.map { mapper.invoke(it) } }
+
+fun <T, R> Flow<Result<T>>.flatMapIfSuccess(mapper: suspend ((T) -> R)): Flow<Result<R>> =
+    flatMapConcat { result -> flow { emit(result.map { mapper.invoke(it) }) } }
```

Comment:
Why is this a flatMap, if what we're calling is just a normal suspend function? Couldn't we just use the `mapIfSuccess` above?